### PR TITLE
Bookmark storing

### LINKED ISSUE #151: Rewrite bookmark storing
So that every character can be used inside bookmark names.

From something like:
```<bookmark_name>
  <desc>here be text</desc>
  <center>37</center>
  <radius>34</radius>
</bookmark_name>```
to 
```<bookmark>
  <name>bookmark name (with possibly weird characters)</name>
  <desc>here be text</desc>
  <center>37</center>
  <radius>34</radius>
</bookmark>```

Fixes #149

### PR DESCRIPTION
**Pull request by mvanmeerten**
Closes #151 #150 #149 #130.

### CODE REVIEW

File: src/main/resources/Basic_Gui.fxml [hunk 1]
```diff
@@ -43,6 +45,7 @@
                    <TextField fx:id="txtMaxDrawDepth" layoutX="-13.0" layoutY="283.0" minWidth="100.0" promptText="Max depth" AnchorPane.leftAnchor="20.0" AnchorPane.rightAnchor="20.0" />
                    <TextField fx:id="txtCenterNode" layoutX="20.0" layoutY="251.0" minWidth="100.0" promptText="Origin node" AnchorPane.leftAnchor="20.0" AnchorPane.rightAnchor="20.0" />
                    <Button fx:id="btnDrawRandom" layoutX="20.0" layoutY="350.0" minWidth="100.0" mnemonicParsing="false" text="Surprise me!" />
+                  <Button fx:id="btnBookmark" layoutX="27.0" layoutY="384.0" mnemonicParsing="false" text="Bookmark" AnchorPane.leftAnchor="20.0" AnchorPane.rightAnchor="20.0" />
```
Comment (YDutchy) [inline src/main/resources/Basic_Gui.fxml]: Is this correctly alligned?

File: src/main/java/programminglife/controller/BookmarkController.java [hunk 2]
```diff
@@ -82,48 +72,32 @@ public static void deleteBookmark(String graphName, String bookMarkName) {
     }
 
     /**
-     * Load a bookmark based on the graph's name and bookmark's name.
-     * @param fileName is the path of the bookmarks file location
-     * @param graphName is the name of the graph in which this bookmark exists.
-     * @param bookMarkName is the name of the bookmark (needs to be unique for bookmarks in the same graph).
-     * @return the {@link Bookmark} you are trying to load.
+     * Checks wheter a bookmark exists.
```
Comment (IwanHoogenboom) [inline src/main/java/programminglife/controller/BookmarkController.java]: Spellfout

File: src/main/java/programminglife/controller/BookmarkController.java [hunk 3]
```diff
@@ -82,48 +72,32 @@ public static void deleteBookmark(String graphName, String bookMarkName) {
...
-                return parseBookmark(el);
+            Element graph = findTag(doc.getElementsByTagName("graph"), graphName);
+            if (graph != null) {
+                Element bookmark = findTag(graph.getElementsByTagName("bookmark"), bookmarkName);
+                if (bookmark != null) {
+                    return true;
+                }
             }
+        } catch (ParserConfigurationException | SAXException | IOException e) {
+            (new Alert(Alert.AlertType.ERROR, "The bookmarks cannot be loaded.", ButtonType.CLOSE)).show();
```
Comment (IwanHoogenboom) [inline src/main/java/programminglife/controller/BookmarkController.java]: This message does not really fit a bookmarkExist method imo.

File: src/main/java/programminglife/controller/BookmarkController.java (hunk 2 above)
Comment (mvanmeerten) [inline src/main/java/programminglife/controller/BookmarkController.java]: spelfout

Comment (IwanHoogenboom) [inline src/main/java/programminglife/controller/BookmarkController.java]: ai, dit is pijnlijk

File: src/main/resources/Basic_Gui.fxml (hunk 1 above)
Comment (mvanmeerten) [inline src/main/resources/Basic_Gui.fxml]: yes!

**Comment by mvanmeerten** [comment]
time spent: 6 hours

**Pull request merged**